src/GafferOSLModule/GafferOSLModule.cpp

```diff
@@ -135,6 +135,19 @@ IECore::CompoundDataPtr shadeWrapper( ShadingEngine &shadingEngine, const IECore
 	return shadingEngine.shade( points, transforms );
 }
 
+boost::python::list getAttributesReadWrapper( ShadingEngine &shadingEngine )
+{
+	boost::python::list list;
+	ShadingEngine::AttributeReadContainer attributeReads =  shadingEngine.getAttributesRead();
+
+	for (ShadingEngine::AttributeReadContainer::const_iterator it = attributeReads.begin(); it != attributeReads.end(); ++it)
+	{
+		boost::python::tuple tuple;
```

Comment:
Let's mirror the C++ with a simple binding of AttributeRead so we don't have to use tuple and guess what each index is on the python side.